### ISSUE TITLE
Should "instrument"/"mechanism" be accessible? keyboard actionable?

### ISSUE THREAD

**Issue by Jym77** [act-rules/act-rules.github.io#1524]
Several of our rules use [instrument](https://act-rules.github.io/glossary/#instrument-to-achieve-an-objective) which is our interpretation of WCAG's [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism).

WCAG's definition has a note stating that:
> The mechanism needs to meet all success criteria for the conformance level claimed.

among other, this means meeting [2.1.1 Keyboard](https://www.w3.org/TR/WCAG21/#keyboard) (be operable by keyboard) and [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) (be included in the accessibility tree) which are both level A.

It seems that whether ours "instrument" need to be included and/or keyboard actionable is a recurring discussion in the related PRs (as far as I can remember). See https://github.com/act-rules/act-rules.github.io/pull/421/files#r537414878 for a recent occurrence of this discussion.

There is little doubt that in the grand scheme of things, the instrument needs to be accessible, but it is less clear that this is required by the individual SC. The note in the definition of "mechanism" is only a note…

---
So:
1. Does WCAG requires that "mechanisms" are accessible (keyboard, screen readers, …)?
2. Should our rules require that "instruments" are accessible?

The two questions are actually disjoint. We can have rules more lenient than WCAG. We can have rules with an assumption that "accessibility of the instrument is needed even if WCAG is not crystal clear on the question".

---
Rules currently using "instrument":
```bash
$ grep -l instrument _rules/*.md
_rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md
_rules/auto-play-audio-has-control-mechanism-4c31df.md
_rules/auto-update-text-efbfc7.md
_rules/device-motion-disabled-c249d5.md
_rules/device-motion-user-interface-7677a9.md
_rules/printable-characters-shortcut-ffbc54.md
```

**Comment by EmmaJP** [comment]
I think the answer to both your questions is yes.

Looking at the WCAG's definition of mechanism, I think the phrase "relied upon" has some significance in regard whether or not WCAG SC should be applied to something not within the webpage. For example, volume could be adjusted by an interactive element within a webpage (to which WCAG applies) or by the device or system controls (to which WCAG doesn't apply). I wouldn't expect a tester to check that device or system controls meet WCAG, even though it's reasonable to rely upon a device or OS having an available mechanism for controlling volume that is in some way accessible to the user.

The ACT Rules definition of "instrument" specifically restricts to something that is within the webpage (testable), so it excludes other mechanisms that are part of the OS, AT or other hardware. The definition requires that rules using it define the objective of the instrument, and that the instrument enables the user to achieve that objective (accessible is somewhat implicit in that). 

I think it would be reasonable to require that rules to define the accessibility expected of the instrument. https://act-rules.github.io/rules/4c31df already does this. Alternatively, or maybe additionally, the definition of "instrument" could include similar expectations that would then apply to all uses of the term.

**Comment by Jym77** [comment]
> I think it would be reasonable to require that rules to define the accessibility expected of the instrument. https://act-rules.github.io/rules/4c31df already does this. Alternatively, or maybe additionally, the definition of "instrument" could include similar expectations that would then apply to all uses of the term.

If we end up deciding that instruments/mechanisms need to comply, I would rather put as much extra conditions as possible in the definition of instrument to limit the risk that they are forgotten in the rules. Notably, all the condition that concern level A criteria can be pushed in the definition since they always apply.

And of course, we can always do checks a minima for the SC we don't really know how to check yet. For example, "included in the accessibility tree + non-empty accessible name" is a minimum check for 4.1.2.

**Comment by Jym77** [comment]
We've decided to keep extra requirements out of the definition to avoid double-flagging the same component, and to pinpoint the failing SC (avoid failing SC about audio because some button is not keyboard usable).
We'll add note in the definition of instrument to explain that.
